### ISSUE TITLE
unreachable code reached

### ISSUE THREAD

**Issue by xNxExOx** [blackbeam/mysql_async#54]
```
thread 'tokio-runtime-worker-1' panicked at 'Couldn't convert Row { member_id: Int(24478), member_group_id: Int(2), mgroup_others: Bytes(""), members_pass_hash: Null, temp_ban: Int(0), warn_count: Int(0), validating: Int(0) } to type (T1, .., T7). (see FromRow documentation)', /home/kubik/.cargo/registry/src/github.com-1ecc6299db9ec823/mysql_common-0.17.0/src/row/convert.rs:368:39
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
thread 'tokio-runtime-worker-2' panicked at 'internal error: entered unreachable code', /home/kubik/.cargo/registry/src/github.com-1ecc6299db9ec823/mysql_async-0.19.0/src/queryable/query_result/mod.rs:167:26
```
I think it is because I expected `members_pass_hash` to be `String`, not `Option<String>`, but I would expect returning error instead of panic, that cause all that it was no longer possible to do any select probably it was not even able to get connection.

**Comment by blackbeam** [comment]
Hi.

> but I would expect returning error instead of panic

Did you use `QueryResult::collect`? It'll panic and i'll provide non-panicing alternative (see #51). 

> that cause all that it was no longer possible to do any select probably it was not even able to get connection.

This was written with an assumption that application will stop immediately after panic (related issue? tokio-rs/tokio#700)

**Comment by xNxExOx** [comment]
Yes I use collect, I did not search for solution even a bit to be honest :( I deployed it in production environment and it crashed 30 minutes later and I need to go somewhere else soon, so I did hotfix, fill the issue and go.
Because I see 'internal error: entered unreachable code' I think it is error in this crate.
The assumption that user **mostly** know the schema is correct, but this schema is not mine and have no not null attributes set, even thou it do not make sense for many columns to be null. So it was actually bug in my code, that I did not expect the null.
Sorry for the issue.
I look forward for not panicking version :)
